### DISCUSSION TITLE
Back at writing characters

### DISCUSSION THREAD

**Discussion by TikkuTokku** [zippity21/Stardew_Valley_Regression_Mod#100] [Ideas]
It's been a while since I contributed to this project. Honestly, writing every character was daunting, and I kind of assumed haunted chocolatier was going to come out sooner, and I wanted to contribute to a mod like this but for that game.

Well, I got the Stardew Bug again, and that immediately translated to writing more characters. This time I'm working with a friend who's going to help me come up with content so I can code it into the game. We already have the archetypes laid out and I am once again excited.

If anyone wants to work with me on writing lines for characters please let me know! I'd love the help.

**Comment by JarOfTar** [comment]
Sure! I've been trying to write a bit of dialogue myself, but always burn out before finishing anything.

**Reply by TikkuTokku** [reply]
Would you like to connect over discord?

**Reply by JarOfTar** [reply]
Sure, hmu on JarOfTar#2995

**Comment by Gooberology** [comment]
Ive got a few things that are work in progress, i really enjoy the tone of the dialogue you write so I've always tried to carry on that wholesome spirit without being too heavy handed on the baby elements.

**Reply by TikkuTokku** [reply]
Thank you! I enjoy that tone as well, and I'd love to hear what you're working on.

I do want a diversity in tone when it comes to the characters, which is why I'm having my friend help me write Sebastian as more of a bully. There's such a huge diversity in the way ABDL is engaged with, and I would love that diversity represented in the mod. So I do plan on having more heavy handed and outright fetishy content through some characters, but I need help with that writing because I'm only really good at writing content that caters to my exact interests.

**Reply by Gooberology** [reply]
Could u give me ur discord so we could talk about it some?

**Reply by TikkuTokku** [reply]
It's the same as my github, and the numbers have been removed

**Reply by Gooberology** [reply]
true, forgot about that

**Comment by RavZenn** [comment]
Ain't exactly have any specific lines in mind, but I'm willing to brainstorm ideas for them and scenarios/ambiance in general. That work for you?

**Reply by TikkuTokku** [reply]
I already have the themes set for each character. I just need help writing the actual lines so I can code them into the game. The writing is the hardest part, especially for themes that are not exactly the way I enjoy experiencing ABDL.

**Reply by RavZenn** [reply]
Think you can post a brief list of the themes and the respective character?

**Reply by TikkuTokku** [reply]
Abigail - Stereotypical Brat
Emily - Dressup Dom
Haley - Mean babysitter
Leah - Uses little space to free artistic mind
Maru - DL - Building robo arm auto changer

Alex - High School jock Bully
Elliot - Very into diapers and writing DL erotica
Harvey - Gentle Parent (I just finished writing him)
Sebastian - Brat focused on teasing about how much you love diapers
Shane - Is prescribed regression therapy and you become his CG

Krobus - Casual DL mostly through curiosity and extremely ace.

**Reply by OriginCosdenex15** [reply]
I tried using the Harvey thing and modify it for the gender swap mod, but I ran into some issues... I managed to get the accident dialogue working, but not matter how high the hearts were, they would always give the "soiled_mean" text. If I were to guess, it has something to do with their name being changed to Harriet... But I don't know why it's always the "soiled_mean" text... I'm not that experienced when it comes to code... I can manage changing the words and can understand faces are the $ next to a number, but this has stumped me...

**Reply by TikkuTokku** [reply]
It's possible that "Harriet" is not a character the mod recognizes. There are fail-safes to give characters without custom dialogue default lines so the game doesn't crash when a line isn't found. It might be doing that.

**Reply by OriginCosdenex15** [reply]
Also an idea to add to Haley it to make her love taking photos of you are playing or just in littlespace. But also don't exactly make her mean, but more so loving to tease you. I just feel like when Haley is closer to you, the initial rudeness wears off quickly, but I feel a more teasing nature fits her better. I still feel like Haley would be a babysitter though.

**Reply by TikkuTokku** [reply]
I like that theming for Haley a lot. Thanks for the suggestion!

**Reply by OriginCosdenex15** [reply]
No problem! I really like this mod and would love to see how the writing is going. I would love to help in testing.

**Reply by TikkuTokku** [reply]
A lot of the additional writing I've done is in collaboration with Goober's additions to the mod which I don't think (I honestly didn't check) are in the latest patch of the mod here. Also, I've taken a hefty pause on writing as my life has gotten busy (full time school and 2 part time jobs). I'm certain I'll get back into it.

**Reply by GardensStarlit** [reply]
Yeah, it looks like Goober's additions are with the previous (non-1.6) version.

If you needed another writer to help pitch in, I'd be happy to assist! 

My schedule is fairly open for the moment, and I've tinkered with dialogue before - (I've added dialogue for a few NPCs in my own run, including full dialogue for Leah, though my take on her leaned more heavily towards the DL side than the AB/Regression theme, so I'd probably start fresh & better match the mod's current tone if I make any submissions for y'all.)

Lmk if you're interested!
((You can add my private discord account, @embers_and_roses if you want, but I _usually_ only check it once a day at around this time if I'm not regularly communicating with anyone on it, so I'd probably see it sooner if you replied here first))

**Reply by TikkuTokku** [reply]
I added you on discord

**Comment by chaos-demon** [comment]
If you still want help writing characters and this is still an ongoing project I'd love to contribute

**Reply by TikkuTokku** [reply]
Oh hello! I haven't done some work on this in a while. I wrote Harvey and the pull request is still up, so I didn't do too much after that. I wouldn't mind helping you write for the mod, though I don't think any pull request will go through at this point.

**Reply by chaos-demon** [reply]
Do you know if this mod is completely abandoned code-wise? I've really been enjoying it and I've started adding dialogue for my own personal playthrough but I suppose it's kind of pointless if the mod will break as soon as the new game update comes out.

**Reply by zippity21** [reply]
I am planning on at least doing a compatibility update, but life has just been going. New job, another kid, etc.

**Reply by chaos-demon** [reply]
That's super valid! I was just worried it would break completely. Thank you so much for keeping this mod running.

**Reply by TikkuTokku** [reply]
Congratulations on another kid! That's really exciting!

**Reply by chaos-demon** [reply]
Man after looking at the code trying to teach myself how to troubleshoot mods in general I am very impressed you've been maintaining this. Am I crazy or is part of this mod coded in binary?

**Comment by Gooberology** [comment]
> Do you know if this mod is completely abandoned code-wise? I've really been enjoying it and I've started adding dialogue for my own personal playthrough but I suppose it's kind of pointless if the mod will break as soon as the new game update comes out.

I should be working on it once SMAPI does its thing. I have a fork i was working on with Tikku before life got in the way and such. I never reached a point where I was satisfied with making a pull request though.

I expect a lot of things to be broken, a lot of methods from the original code were depreciated, and support for them would be removed with the next major version of SMAPI, which I have no reason to not expect its commin with 1.6.

Tbh i still have a lot going on, but I do really care about this mod. Ill see what I can do.

**Reply by zippity21** [reply]
I've started the update, but I think I'm just waiting for PyTk to update. [zippity21/1p6_compatability](https://github.com/zippity21/Stardew_Valley_Regression_Mod/tree/zippity21/1p6_compatability)

**Reply by Gooberology** [reply]
Alright. Let me know if you need anything, and i'll work on trying to incorporate my changes
